iroha/src/block.rs

```diff
@@ -122,9 +123,17 @@ impl ChainedBlock {
                 Err(e) => log::warn!("Transaction validation failed: {}", e),
             }
         }
-        //TODO: rebuild merkle tree and reassign `merkle_root_hash`, as tx set may be different.
+        let mut merkle_tree = MerkleTree::new();
+        merkle_tree.build(
+            &transactions
+                .iter()
+                .map(|transaction| transaction.hash())
+                .collect::<Vec<_>>(),
+        );
+        let mut header = self.header;
+        header.merkle_root_hash = merkle_tree.root_hash();
```

Comment:
Because we do not log and do not use this variables, maybe combine all operations in one?
`self.header.merkle_root_hash = MerkleTree::new().build(...).root_hash();`

Reply:
Do we even need `MerkleTree::new` in this case?